src/librustc_mir/build/block.rs

```diff
@@ -113,7 +114,8 @@ impl<'a, 'gcx, 'tcx> Builder<'a, 'gcx, 'tcx> {
                     // Declare the bindings, which may create a visibility scope.
                     let remainder_span = remainder_scope.span(this.hir.tcx(),
                                                               &this.hir.region_scope_tree);
-                    let scope = this.declare_bindings(None, remainder_span, lint_level, &pattern);
+                    let scope = this.declare_bindings(None, remainder_span, lint_level, &pattern,
+                                                      false);
```

Comment:
Nit: a plain `false` like this is always a bit unclear to me as a reader. Maybe we could put a newtype here, like `declare_bindings(..., InGuard(false))` (and `struct InGuard(bool)`)?

Reply:
Or I should probably just reuse the `enum` I introduced somewhere else in this PR...

Reply:
(well, technically this boolean flag has a different meaning currently than the `WithinGuard`/`OutsideGuard` enum... but I'll figure something out. I too dislike plain booleans.)